Structure_Engine/Create/PanelPlanar.cs

```diff
@@ -77,40 +79,34 @@ public static PanelPlanar PanelPlanar(List<Edge> externalEdges, List<Opening> op
 
         /***************************************************/
 
+        [DeprecatedAttribute("Generic method for ICurve in place")]
         public static List<PanelPlanar> PanelPlanar(List<Polyline> outlines, ISurfaceProperty property = null, string name = "")
         {
-            List<PanelPlanar> result = new List<PanelPlanar>();
-            List<List<Polyline>> sortedOutlines = outlines.DistributeOutlines();
-            foreach (List<Polyline> panelOutlines in sortedOutlines)
-            {
-                List<Edge> externalEdges = panelOutlines[0].SubParts().Select(o => new Edge { Curve = o }).ToList();
-                List<Opening> openings = new List<Opening>();
-                foreach (Polyline p in panelOutlines.Skip(1))
-                {
-                    List<Edge> openingEdges = p.SubParts().Select(o => new Edge { Curve = o }).ToList();
-                    openings.Add(new Opening { Edges = openingEdges });
-                }
-                result.Add(new PanelPlanar { ExternalEdges = externalEdges, Openings = openings, Property = property, Name = name });
-            }
-            return result;
+            return PanelPlanar(new List<ICurve>(outlines), property, name);
         }
 
         /***************************************************/
 
         public static List<PanelPlanar> PanelPlanar(List<ICurve> outlines, ISurfaceProperty property = null, string name = "")
         {
             List<PanelPlanar> result = new List<PanelPlanar>();
-            List<List<ICurve>> sortedOutlines = outlines.DistributeOutlines();
-            foreach (List<ICurve> panelOutlines in sortedOutlines)
+            List<List<IElement1D>> outlineEdges = outlines.Select(x => x.ISubParts().Select(y => new Edge { Curve = y } as IElement1D).ToList()).ToList();
+            
+            List<List<List<IElement1D>>> sortedOutlines = outlineEdges.DistributeOutlines();
+            foreach (List<List<IElement1D>> panelOutlines in sortedOutlines)
             {
-                List<Edge> externalEdges = panelOutlines[0].ISubParts().Select(o => new Edge { Curve = o }).ToList();
-                List<Opening> openings = new List<Opening>();
-                foreach (ICurve p in panelOutlines.Skip(1))
+                PanelPlanar panel = new PanelPlanar();
+                panel = panel.SetOutlineElements1D(panelOutlines[0]);
+                List<IElement2D> openings = new List<IElement2D>();
```

Comment:
Think this could be simplified/clarified to:

```
                PanelPlanar panel = new PanelPlanar();
                panel = panel.SetOutlineElements1D(panelOutlines[0]);
                List<Opening> openings = new List<Opening>();
                foreach (List<IElement1D> p in panelOutlines.Skip(1))
                {
                    Opening opening = (new Opening()).SetOutlineElements1D(p);
                    openings.Add(opening);
                }
                panel.Openings = openings;
                panel.Property = property;
                panel.Name = name;
                result.Add(panel);
```

Reply:
You are right, commited the above.